proof-systems/src/darlin/tests/mod.rs

```diff
@@ -142,14 +144,38 @@ mod test {
 
         // Since the AHP failed, we are able to determine which proof verification has failed
         assert_eq!(result.unwrap_err().unwrap(), idx);
+
+        if fake_pcd.is_some() {
+
+            // Restore correct PCD
+            pcds[idx] = original_pcd;
+
+            let idx: usize = rng.gen_range(0, pcds.len());
+            pcds[idx] = fake_pcd.unwrap();
+            pcds[idx].randomize_proof(rng);
+
+            let result = verify_aggregated_proofs::<G1, G2, D, R>(
+                pcds,
+                vks,
+                &proof_g1,
+                &proof_g2,
+                verifier_key_g1,
+                verifier_key_g2,
+                rng
+            );
+
+            // Check AHP failed
+            assert!(result.is_err());
```

Comment:
Can you assert also that the Option returned as Error is None ? One of the concern was to have a test that fails without being able to return the index of the failing proof (e.g. failing in the hard part)

Reply:
is_none() check added